src/dmd/dmodule.d

```diff
@@ -1356,3 +1383,232 @@ struct ModuleDeclaration
+
+    if (info.depth == 0)
+        printf(".");
+    else
+    {
+        foreach (i; 0..info.depth)
+        {
+            if (i > 0) printf(".");
+            printf("%s", matchNodes[matcherIndex + i].id.toChars);
+        }
+    }
+}
+
+private void createMatchNodes()
+{
+    static size_t findSortedIndexToAddForDepth(size_t depth)
+    {
+        size_t index = 0;
+        while (index < matchNodes.dim)
+        {
+            auto info = matchNodes[index];
+            if (depth > info.depth)
+                break;
+            index += 1 + info.depth;
+        }
+        return index;
+    }
+
+    if (matchNodes.dim == 0)
+    {
+        for (size_t packageIndex = 0; packageIndex < global.params.includeMatchStrings.dim; packageIndex++)
+        {
+            auto matchString = global.params.includeMatchStrings[packageIndex];
+            auto depth = parseMatchStringDepth(matchString);
+            auto entryIndex = findSortedIndexToAddForDepth(depth);
+            matchNodes.split(entryIndex, depth + 1);
+            parseMatchString(matchString, &matchNodes[entryIndex], depth);
+            // if at least 1 "include pattern" is given, then it is assumed the
+            // user only wants to include modules that were explicitly given, which
+            // changes the default behavior from inclusion to exclusion.
+            if (includeByDefault && !matchNodes[entryIndex].isExclude)
+            {
+                //fprintf(global.stdmsg, "Matcher: found 'include pattern', switching default behavior to exclusion\n");
+                includeByDefault = false;
+            }
+        }
+
+        // Add the default 1 depth matchers
+        MatcherNode[8] defaultDepth1MatchNodes = [
+            MatcherNode(true, 1), MatcherNode(Id.std),
+            MatcherNode(true, 1), MatcherNode(Id.core),
+            MatcherNode(true, 1), MatcherNode(Id.etc),
+            MatcherNode(true, 1), MatcherNode(Id.object),
+        ];
+        {
+            auto index = findSortedIndexToAddForDepth(1);
+            matchNodes.split(index, defaultDepth1MatchNodes.length);
+            matchNodes.data[index .. index + defaultDepth1MatchNodes.length] = defaultDepth1MatchNodes[];
+        }
+    }
+}
+
+private bool endOfPattern(char c) { return c == ',' || c == '\0'; }
+
+/** Parses a 'match string', which is the "include import" components
+    given on the command line, i.e. "-i=<match_string>,<match_string>,..."
+*/
+private ushort parseMatchStringDepth(const(char)* matchString)
+{
+    if (matchString[0] == '-')
+        matchString++;
+
+    // handle special case
+    if (matchString[0] == '.' && endOfPattern(matchString[1]))
+        return 0;
+
+    ushort depth = 1;
+    for (;; matchString++)
+    {
+        auto c = *matchString;
+        if (c == '.')
+            depth++;
+        if (endOfPattern(c))
+            return depth;
+    }
+}
+private void parseMatchString(const(char)* matchString, MatcherNode* dst, ushort depth)
```

Comment:
please add a DDoc header for all functions that include a summary, `Params` and `Returns`.

Reply:
@JinShil traditionally we don't ddoc private functions, but we definitely should document them.